src/test/java/picard/sam/FilterSamReadsTest.java

```diff
@@ -217,7 +218,45 @@ public void testPairedIntervalFilter(final String intervalFilename, final int ex
         Assert.assertEquals(count, expectNumber);
     }
 
-    private FilterSamReads setupProgram(final File inputFile, final File inputSam, final FilterSamReads.Filter filter) throws Exception {
+    @DataProvider(name = "dataTestDebugOption")
+    public Object[][] dataTestDebugOption() {
+        return new Object[][]{
+                {null, false},
+                {true, true},
+                {false, false}
+        };
+    }
+
+    /**
+     * makes sure debug files are created properly
+     */
+    @Test(dataProvider = "dataTestDebugOption")
+    public void testDebugOption(Boolean writeDebugReads, Boolean isDebugFileExpected) throws Exception {
+        // input as SAM file
+        final File inputSam = new File("testdata/picard/sam/aligned.sam");
+        final File javascriptFile = new File("testdata/picard/sam/FilterSamReads/filterOddStarts.js");
+
+        FilterSamReads filterTest = setupProgram(javascriptFile, inputSam, FilterSamReads.Filter.includeJavascript, writeDebugReads);
+        Assert.assertEquals(filterTest.doWork(), 0);
+
+        final File inputReadsFile = new File(filterTest.OUTPUT.getParentFile(), IOUtil.basename(filterTest.INPUT) + ".reads");
+        Assert.assertEquals(inputReadsFile.exists(), isDebugFileExpected.booleanValue());
+
+        final File outputReadsFile = new File(filterTest.OUTPUT.getParentFile(), IOUtil.basename(filterTest.OUTPUT) + ".reads");
+        Assert.assertEquals(outputReadsFile.exists(), isDebugFileExpected.booleanValue());
+
+        // We have to clean up the debug files after each test is run to make sure a clean state is preserved in between tests
+        // This mostly affects the input *.reads file because it will always be called "aligned.reads" and will cause future
+        // tests to fail if it sticks around and we dont expect it to be written
+        if (inputReadsFile.exists()) {
```

Comment:
use Files.deleteIfExists()

Reply:
This actually doesn't work in this case because it wont try to delete the file until all runs of the `testDebugOption` method complete but I need it to delete the files between runs.  At least that is what I think it is doing because I tried that and on my 3rd test always failed because the the `aligned.reads` file still existed from the second test where it is supposed to be written out.

Reply:
oh...I read your comment incorrectly.  I read `deleteOnExit()`....Sorry!  It took me forever to figure out why `deleteOnExit()` wasnt working